### ISSUE TITLE
Surfaces will be invisible

### ISSUE THREAD

**Issue by Hisashimaru** [LogicalError/realtime-CSG-for-unity#251]
Surfaces will be invisible, If you turn off "cast shadows" or "receive shadows" in the surfaces settings.

**Comment by LogicalError** [comment]
This seems to be because of deferred or forward rendering (I forgot which one). In one of those modes the whole mesh becomes invisible when you turn off "cast shadows" or "receive shadows" on a MeshRenderer ..

**Comment by thevaber** [comment]
I noticed the same issue, it doesn't seem to be related to using deferred/forward though. Only two combinations of Visible/Receive/Cast seem to actually generate render meshes, Visible+Receive+Cast and Visible-only (Collision flag doesn't have any effect on this); the remaining combinations only create helper meshes in the editor. I tried a couple of versions and first version where this problem appears seems to be 1.517b (which also removed some shadow related settings from CSG Model -> Rendering). With 1.516, render meshes are generated properly for all the unique combinations of the three flags.
